### PR TITLE
feat(reactor): remove usage of time.sleep cause it was big ghetto on god fr fr

### PR DESCRIPTION
**Pull request by itsdevbear**
The PR removes the usage of time.Sleep within the consensus reactor.

### CODE REVIEW

**Comment by cason** [comment]
Why?

**Comment by itsdevbear** [comment]
time.sleep is a bit messy and there is a gothonic solution to this design using select statements

File: internal/consensus/reactor.go [hunk 1]
```diff
@@ -728,15 +726,10 @@ OUTER_LOOP:
 							BlockID: maj23.ToProto(),
 						},
 					})
-					time.Sleep(conR.conS.config.PeerQueryMaj23SleepDuration)
```
Comment (melekes) [inline internal/consensus/reactor.go:731]: to preserve the old behavior we must `Reset` ticker here

File: internal/consensus/reactor.go [hunk 2]
```diff
@@ -635,88 +637,84 @@ OUTER_LOOP:
...
-
-		time.Sleep(conR.conS.config.PeerGossipSleepDuration)
 	}
 }
 
 // NOTE: `queryMaj23Routine` has a simple crude design since it only comes
 // into play for liveness when there's a signature DDoS attack happening.
 func (conR *Reactor) queryMaj23Routine(peer p2p.Peer, ps *PeerState) {
-OUTER_LOOP:
+	ticker := time.NewTicker(conR.conS.config.PeerQueryMaj23SleepDuration)
```
Comment (melekes) [inline internal/consensus/reactor.go:703]: I actually like this part ❤️

File: internal/consensus/reactor.go [hunk 3]
```diff
@@ -635,88 +637,84 @@ OUTER_LOOP:
...
+	defer ticker.Stop()
+
 	for {
```
Comment (melekes) [inline internal/consensus/reactor.go:706]: Should we tick immediately to preserve the old behavior?

**Review by melekes — COMMENTED** [review]
I appreciate the effort here, but tickers make this code less predictable. I don't think there are any immediate downsides to ticking more often, but I'd rather keep the old code if there are no significant performance gains.

**Pull request closed without merging**